src/julia.h

```diff
@@ -1492,6 +1495,8 @@ STATIC_INLINE void jl_eh_restore_state(jl_handler_t *eh)
     }
 #endif
     JL_SIGATOMIC_END();
+    jl_finalizers_enable(eh->finalizers_enabled);
+    jl_gc_enable(eh->gc_enabled);
```

Comment:
I think gc is OK, I mainly didn't want finalizers or other non-sigatom regions (because they can corrupt `jl_exception_in_transit` + the `catch_backtrace()`, or and possible also skip `finally` blocks). i'm starting to think we need to make a dedicated thread for running all finalizers.